### DISCUSSION TITLE
Is it possible to create a transcoder plugin adding support for new formats?

### DISCUSSION THREAD

**Discussion by ZzZombo** [jellyfin/jellyfin#11058] [Q&A]
And while at that, can I see the complete list of interfaces intended to be implemented by plugins in order to extend/provide new functionality?

**Comment by gnattu** [comment]
The transcoding flow workflow is tightly integrated with ffmpeg, and it would be hard to add "plugins" for now.

Do you have anything specific you want to do in mind? Maybe we can find out something we can do for now.

**Reply by ZzZombo** [reply]
I want to extend the available image formats, including RAW and HEIC.

**Reply by nyanmisaka** [reply]
AFAIK HEIC is not yet supported before FFmpeg 7.0.

**Reply by gnattu** [reply]
If it is only pictures, we *may* be able to extend our pictures to use a webhook interface, as a picture is nothing more than a static file and can be processed by web-browsers easily.

The workflow would be:

- The "external handler" start listening on an http port, let's say `127.0.0.1:9999`
- When Jellyfin received the user requesting an image file it cannot handle, it forwards the image to `127.0.0.0:9999`
- Then the handler returns a converted image, let's say png to Jellyfin
- Jellyfin responds user with that converted png.

The external handler may specify the file type it want to handle, like `*.RAW`, `*.HEIC`.

**Reply by ZzZombo** [reply]
> AFAIK HEIC is not yet supported before FFmpeg 7.0.

Can't one call a different instance of it just for this task?

**Reply by ZzZombo** [reply]
Guys, anything concrete for me to go by?

**Reply by JPVenson** [reply]
FYI:
Long ago i formulated as read-to-implement design that would allow this:
https://github.com/jellyfin/jellyfin-meta/discussions/36
